src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -146,7 +162,16 @@ private static Condition handleNotEquals(NotEqualsCondition notEqualsCondition,
       return handleDate(notEqualsCondition, field);
     }
     String dataType = getFieldDataType(entityType, notEqualsCondition);
-    if (STRING_TYPE.equals(dataType) || RANGED_UUID_TYPE.equals(dataType) || OPEN_UUID_TYPE.equals(dataType) || DATE_TYPE.equals(dataType)) {
+    String filterFieldDataType = getFieldForFiltering(notEqualsCondition, entityType).getDataType().getDataType();
+    if (RANGED_UUID_TYPE.equals(filterFieldDataType) || OPEN_UUID_TYPE.equals(filterFieldDataType) ) {
+      String value = (String) notEqualsCondition.value();
+      if (!isValidUUID(value)) {
+        Map<String, String> errors = Map.of(INVALID_UUID, ERROR_MESSAGE + value);
+        throw new InvalidFqlException(field + notEqualsCondition.toString(), errors);
+      }
+      return cast(field, UUID.class).eq(cast(value, UUID.class));
+    }
+    if (STRING_TYPE.equals(dataType) || RANGED_UUID_TYPE.equals(dataType) || OPEN_UUID_TYPE.equals(dataType) || DATE_TYPE.equals(dataType) || STRING_UUID_TYPE.equals(dataType)) {
```

Comment:
We don't need the RANGED_UUID_TYPE or OPEN_UUID_TYPE conditions here anymore, do we? Since they'll be handled above.